core/src/com/pottda/game/physicsBox2D/CollisionListener.java

```diff
@@ -0,0 +1,72 @@
+package com.pottda.game.physicsBox2D;
+
+import com.badlogic.gdx.physics.box2d.*;
+import com.pottda.game.model.*;
+import com.pottda.game.model.Character;
+
+public class CollisionListener implements ContactListener {
+    @Override
+    public void beginContact(Contact contact) {
+        // Continue if one fixture is a sensor and one isn't
+        if (contact.getFixtureA().isSensor() == contact.getFixtureB().isSensor()) {
+            return;
+        }
+
+        Body projectileBody;
+        Body targetBody;
+
+        // Set bodies to their corresponding variable, only Projectiles have sensors
+        if (contact.getFixtureA().isSensor()) {
+            projectileBody = contact.getFixtureA().getBody();
+            targetBody = contact.getFixtureB().getBody();
+        } else {
+            projectileBody = contact.getFixtureB().getBody();
+            targetBody = contact.getFixtureA().getBody();
+        }
+
+        Projectile projectile = projectileBody.getUserData() instanceof Projectile ?
+                ((Projectile) projectileBody.getUserData()) : null;
+        Character target = targetBody.getUserData() instanceof Character ?
+                ((Character) targetBody.getUserData()) : null;
+
+        // Make sure projectile actually IS a Projectile
+        if(projectile == null){
```

Comment:
This check could be above line 29, because if isn't a projectile we probably don't care about what it's colliding with.